### PR TITLE
Lesion Detection in ISIC 2018 using YOLOv11 - 48022835

### PR DESCRIPTION
**Pull request by ETraffordUQ**
Project repository for model designed for detection of lesions in the ISIC2018 dermoscopic images dataset by fine-tuning a pre-trained Ultralytics YOLOv11 model. https://challenge.isic-archive.com/data/#2018

The model has been fine-tuned to improve accuracy in identifying skin lesions from medical images, leveraging YOLOv11’s fast and efficient detection capabilities. Key enhancements include customized bounding box predictions for lesions, along with evaluation metrics such as IoU for performance tracking on the test set. The integration aims to support high-speed and accurate lesion detection, critical for medical imaging applications.

**Provides the following:**
```README.md```: Documentation regarding project usage, outcomes and information.
```dataset.py```: Processes the ISIC2018 Task 1 Datasets for interaction with YOLOv11 model.
```modules.py```: Pre-installs commonly used weights and allows definition of a new model.
```train.py```: Trains the YOLOv11 model on the training and validation data, with option to run testing on test data.
```predict.py```: Runs inference on given images or entire test set to indicate model performance.
```utils.py```:  Adjustable paths for downloaded raw ISIC2018 Task 1 Datasets.
```README_figures/```: folder containing images and plots used in the README.md file

- Ewan Trafford, SID 48022835

### CODE REVIEW

**Comment by shaivikaaaa** [comment]
**This is an initial inspection**

Difficulty : Normal
Task : 3

1) Recognition Problem:

- code structure is good
- graph plot are shown
- images are shown - that is good 
- file structure is ok

2) Commit Log:

- Commit message are good
- Commits are progressive 

3) Documentation:
- do not have a section of "Reproducibility" stating how can someone else can reproduce the results and run the model to achieve same results

4) Pull Request

- No result section in PR

Feedback:

- add a section of "Reproducibility" stating how can someone else can reproduce the results and run the model to achieve same results
-  mention train/test % achieved in PR

**Comment by aniketgupta17** [comment]
Observational Feedback

Pull Request:
Correctly created the Pull request from Topic Recognition Branch .
The pull request includes clear description about the file structure but missing results .
NO Previous feedback incorporated .

File Organizing: Well-organized files.

Commit Log:
Commit messages are progressive for the Recognition Problem solved using 4 files .
Commits show a logical developement .

Documentation:
Code comments and docstrings are included.
Proper GitHub markdown formatting is used, with organized headings, and lists .
Can add about Conclusion and future improvements in the ReadMe.

**Comment by ETraffordUQ** [comment]
Hello tutors, I am confused on the process of implementing the feedback from the initial inspection above. I was under the impression (from ED Discussion post #504, and general lack of answers in other relevant posts) that I was not to make any changes to my project or pull request until Shakes did his marking and outlined if there was anything I could change for a few extra marks. I do however note in @aniketgupta17 's comment, they stated "No previous feedback incorporated", one day after the initial inspection was done. 
Please help clarify this for me: am I expected to implement the feedback from @shaivikaaaa 's initial inspection before or after Shakes does his marking, and if I am supposed to do it before, is it now too late?

**Comment by shaivikaaaa** [comment]
Hi @ETraffordUQ 
The feedback which I gave you is based on minor changes like adding few lines in your README or just modifying your PR a bit.
You CANNOT do any MAJOR changes related to codebase or anything else.
The sole reason for these initial inspection is to look over if the student has initiated the PR correctly, or mentioned everything asked for

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>Header blocks. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>Good solution to problem. </td><td></td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting present. </td><td></td></tr>
	<tr><td></td><td>No Data leakage found. </td><td></td></tr>
	<tr><td></td><td>Difficulty : Normal. YOLO (Normal Difficulty)</td><td>-5</td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Good Meaningful commit messages. </td><td></td></tr>
	<tr><td></td><td>Good Progressive commits. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Good. </td><td></td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Good. </td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF Checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). On time.</td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is good. </td><td></td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-5</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_
